Production/src/BaseTupleProducer.cc

```diff
@@ -960,14 +960,16 @@ void BaseTupleProducer::FillEventTuple(const analysis::SelectionResultsBase& sel
             eventTuple().fatJets_jettiness_tau1.push_back(GetUserFloat(jet, "NjettinessAK8:tau1"));
             eventTuple().fatJets_jettiness_tau2.push_back(GetUserFloat(jet, "NjettinessAK8:tau2"));
             eventTuple().fatJets_jettiness_tau3.push_back(GetUserFloat(jet, "NjettinessAK8:tau3"));
-        } else if(period == Period::Run2017) {
+        } else if(period == Period::Run2017 ||period == Period::Run2018 ) {
             subjets_collection = "SoftDropPuppi";
             eventTuple().fatJets_m_softDrop.push_back(GetUserFloat(jet, "ak8PFJetsPuppiSoftDropMass"));
             eventTuple().fatJets_jettiness_tau1.push_back(GetUserFloat(jet, "NjettinessAK8Puppi:tau1"));
             eventTuple().fatJets_jettiness_tau2.push_back(GetUserFloat(jet, "NjettinessAK8Puppi:tau2"));
             eventTuple().fatJets_jettiness_tau3.push_back(GetUserFloat(jet, "NjettinessAK8Puppi:tau3"));
             eventTuple().fatJets_jettiness_tau4.push_back(GetUserFloat(jet, "NjettinessAK8Puppi:tau4"));
         }
+        else
+            throw analysis::exception("period not allowd.");
```

Comment:
`throw analysis::exception("period not allowd.");` -> `throw analysis::exception("FillEventTuple: fill fatJets for period %1%" is not implemented") % period;`